### DISCUSSION TITLE
How to change the SQL executed by 「edit data」button

### DISCUSSION THREAD

**Discussion by tttangz** [dbeaver/dbeaver#20244] [Q&A] [answered]
![sample](https://github.com/dbeaver/dbeaver/assets/114927200/4d83ffc1-eb6c-43f9-ac4f-9fc15cf58f36)

When I click the “edit data” button, the SQL executed is “SELECT * FROM EGC_DB.EASCFM.EATBFM”，
This will result in an incorrect result，
The SQL I want to execute is “SELECT * FROM EASCFM.EATBFM”. (without EGC_DB)
Is there any way to achieve this？

**Comment by E1izabeth** [comment]
You can use `Read data in SQL Editor` button to read data in SQL editor and then correct the query there.

What database do you use?

**Reply by tttangz** [reply]
Thank you for your reply. :) 
Database is Syfmoware,it is not a mainstream database，But our company is using it.
`Read data in SQL Editor` can work for me, but i will lose the function of  the `edit data`  button.
Does the current version not support rewriting the query parameters of  the `edit data` button？
（Although in most cases no modifications are required）

**Reply by E1izabeth** [reply, marked as answer]
Usually, it's supported. Probably you need to configure your generic driver parameters https://dbeaver.com/docs/wiki/Database-drivers/

**Reply by E1izabeth** [reply]
I hope with proper configuration usual table opening might start working.

**Reply by tttangz** [reply]
Thank you! I will try it on weekdays. good luck for us. :)

**Reply by tttangz** [reply]
https://dbeaver.com/docs/wiki/Database-drivers/#main-parameters
`Omit catalog(s) `
This option can omit the database name，now  the table can be opened normall.
That's great. 👍
